staging/src/k8s.io/client-go/testing/fixture.go

```diff
@@ -482,6 +483,28 @@ func filterByNamespaceAndName(objs []runtime.Object, ns, name string) ([]runtime
 	return res, nil
 }
 
+// filterList returns all objects in the collection matched by the namespace and the provided ListRestrictions/Selectors
+// Only filter by label.Selector since arbitrary field selectors are not yet supported in K8s
+func filterList(objs []runtime.Object, ns string, lr ListRestrictions) ([]runtime.Object, error) {
+	var res []runtime.Object
+
+	for _, obj := range objs {
+		acc, err := meta.Accessor(obj)
+		if err != nil {
+			return nil, err
+		}
+		if ns != "" && acc.GetNamespace() != ns {
+			continue
+		}
+		if !lr.Labels.Matches(labels.Set(acc.GetLabels())) {
```

Comment:
since that we pass the whole `ListRestrictions` in, can we make field selector work too?

Reply:
IIUC since https://github.com/kubernetes/kubernetes/issues/53459 is not resolved, it might be confusing to have an actual client that is unable to filter by arbitrary fields, but a fake client that can. I passed in `ListRestrictions` so that it would be simple enough to add when the functionality is added. I can add this in now if you think it makes sense to.